### DISCUSSION TITLE
v2.0.38b1

### DISCUSSION THREAD

**Discussion by DTTerastar** [ESPresense/ESPresense#381] [Announcements]
## Changes

Disable unused ble functions to free up some memory
Added Count of devices present in room
Added Known Macs configuration
Fix trailing space matching everything
Move querying to separate task
Tweak query parameters to make more stable

<hr /><em>This discussion was created from the release <a href='https://github.com/ESPresense/ESPresense/releases/tag/v2.0.38b1'>v2.0.38b1</a>.</em>

**Comment by erwinSmith757** [comment]
this sounds great, i really like the count of devices in a room part, what would be even better, if you can mark known devices....so it can be used for example activate guest mode...

**Comment by DTTerastar** [comment]
The best part is we now have around 100 K of free memory. So auto updating should work without the razor thin margins of before!

**Reply by DTTerastar** [reply]
<img width="1459" alt="image" src="https://user-images.githubusercontent.com/1491145/157653571-f3890579-ff0a-4cf6-a7bb-e98bd66bb60f.png">

**Reply by markmghali** [reply]
seriously thank you for all your hard work! I am loving these sensors!

**Reply by erwinSmith757** [reply]
every release im so amazed of the progress.... this is one of the best projects ive ever followed....thank you for all this great work!!!!

**Comment by Zipties** [comment]
Room counting sounds interesting. How are the "rooms" setup on the HA side?

**Reply by DTTerastar** [reply]
Once you configure it, it shows up like the availability topic using auto discovery: 
![image](https://user-images.githubusercontent.com/1491145/157896200-9803fd4f-c572-46e2-affa-97e098a1d566.png)

**Comment by DTTerastar** [comment]
Right now it's just a count on telemetry which only updates every 15 seconds, so it's slow.  It also has some issues w/ random macs rotating making the count go down for a bit.  I'm thinking through how the algo should work to be resistant to rotating macs.
